templates/xpress/main.py

```diff
@@ -0,0 +1,126 @@
+"""
+Template for working with FICO Xpress.
+"""
+
+import argparse
+import json
+import sys
+from typing import Any, Dict
+
+import xpress as xp
+
+
+def main() -> None:
+    """Entry point for the template."""
+
+    parser = argparse.ArgumentParser(description="Solve problems with Xpress.")
+    parser.add_argument(
+        "-input",
+        default="",
+        help="Path to input file. Default is stdin.",
+    )
+    parser.add_argument(
+        "-output",
+        default="",
+        help="Path to output file. Default is stdout.",
+    )
+    parser.add_argument(
+        "-duration",
+        default=30,
+        help="Max runtime duration (in seconds). Default is 30.",
+        type=int,
+    )
+    args = parser.parse_args()
+
+    # Read input data, solve the problem and write the solution.
+    input_data = read_input(args.input)
+    solution = solve(input_data, args.duration)
+    write_output(args.output, solution)
+
+
+def solve(input_data: Dict[str, Any], duration: int) -> Dict[str, Any]:
+    """Solves the given problem and returns the solution."""
+
+    # Creates the problem.
+    problem = xp.problem()
+    problem.setControl("timelimit", duration)
+
+    # Initializes the linear sums.
+    weights = 0.0
+    values = 0.0
+
+    # Creates the decision variables and adds them to the linear sums.
+    items = []
+    for item in input_data["items"]:
+        item_variable = xp.var(vartype=xp.binary, name=item["item_id"])
```

Comment:
Is it too late to change the inputs of the knapsack models from `items.item_id` (stutters) to `items.id`? If it bugs you like me, go for it. However, I also didn't take the time, so, feel free to leave it.
Thanks for keeping them in one format though! :hugs:

Reply:
Consider it done.